### DISCUSSION TITLE
Как создавать Custom fields (Произвольные поля)?

### DISCUSSION THREAD

**Discussion by nvda-ru** [maxsite/cms#450] [Q&A]
Приветствую всех.
Подскажите пожалуйста, как в Maxsite создавать различные произвольные поля?
 Нужно к основному полю для текста в записи, добавить поля для вставки ссылок дна источник новости, автора, возможно ещё какие-то текстовые данные.
Вот здесь есть хорошее описание о чём речь:
https://n-wp.ru/udobnaya-rabota-s-proizvolnyimi-polyami-s-pomoshhyu-plagina-maxsite-proizvolnyie-polya/

Искал такое дополнение для Maxsite, но найти не удалось.

В документации
https://max-3000.com/doc
прочитал эти статьи:
6. Каталог custom

https://max-3000.com/doc/template-custom
12. Мета опции записи
https://max-3000.com/doc/template-meta

Только, не понял каким образом составить код для произвольного текстового поля, где будет поле для ссылки и поле для анкора ссылки.
Поле для однострочного текста и поле для многострочного.

Такого образца не обнаружил.
И после, каким образом подключить эти поля чтобы данные выводились  в нужных местах записи.
А если добавленные поля не заполнялись, то их не выводить.

На текущий момент, на сайте версия 107.4, ещё не обновлял до 108.
Благодарю за информацию, образцы.

**Comment by maxsite** [comment]
Открываете файл `custom/my_meta.ini` и в нём в формате ini-файла задаются опции для записи.  Например:

```
[CSS-класс записи]
options_key = mso-page-content-add-class
description = "Будет добавлен к основому контейнеру"
type = textfield
default = ""
```
Вот ещё есть описание: https://max-3000.com/page/maxsite-cms-dlja-verstalshhikov-5
В старой версии MaxSite CMS был такой файл: https://github.com/maxsite/cms/blob/v086/application/maxsite/common/ini%20-%20readme.txt

**Comment by nvda-ru** [comment]
Максим благодарю за пояснение.
Да, я эти ini файлы смотрел, но фраза meta меня ввела в заблуждение.
Искал именно Custom fields, поэтому решил что это что-то другое, хоть и похожее на то что мне нужно.

А через поиск поискать материалы по "Метаполям" не догадался.

 В документации:
https://max-3000.com/page/maxsite-cms-dlja-verstalshhikov-5
много для себя почерпнул, да кроме этого, там ещё прочитал сопутствующие материалы.
Буду разбиратся, благо у вас там есть образцы с пояснениями.
